app/lib/usecase/optics/generate_jwt_token.rb

```diff
@@ -0,0 +1,29 @@
+module Usecase
+  module Optics
+    class GenerateJwtToken
+      def initialize(url:, api_key:, hmac_secret:)
+        @url = url
+        @api_key = api_key
+        @hmac_secret = hmac_secret
+      end
+
+      def execute
+        JWT.encode payload, hmac_secret, 'HS256'
+      end
+
+      private
+
+      def payload
+        {
+          iss: api_key,
+          aud: url,
+          iat: Time.now.to_i,
+          exp: Time.now.to_i + ONE_HOUR_IN_SECONDS
+        }
+      end
+
+      attr_reader :url, :api_key, :hmac_secret
+      ONE_HOUR_IN_SECONDS = 3600 * 3600
```

Comment:
don't think this is quite right 😄

Reply:
😆

Reply:
Have removed the hardcoded expiry time.  Not useful to us as this is the default and I doubt we would rely on time to determine when to request a new one.